scripts/build.py

```diff
@@ -174,47 +181,127 @@ def _ensure_files_exist(filepaths):
         filepaths: list(str). Paths to files that we want to ensure exist.
 
     Raises:
-        Exception: Raised if one or more of the files does not exist.
+        OSError: One or more of the files does not exist.
     """
     for filepath in filepaths:
         if not os.path.isfile(filepath):
-            raise Exception('File %s does not exist.' % filepath)
+            raise OSError('File %s does not exist.' % filepath)
 
 
-def process_html(source_path, target_path, file_hashes):
-    """Copies contents of HTML file, while removing whitespace and
-    replacing paths inside the file with paths with hashes.
+def get_file_count(directory_path):
+    """Count total number of file in directory, ignoring any files with
+    extensions in FILE_EXTENSIONS_TO_IGNORE.
 
     Args:
-        source_path: str. Absolute path to file to be processed.
-        target_path: str. Absolute path to location where to copy
-            the processed file.
-        file_hashes: dict(str, str). Dictionary of file hashes.
+        directory_path: str. Directory to be walked.
+
+    Returns:
+        int. Total number of files minus ignored files.
+    """
+    total_file_count = 0
+    for _, _, filenames in os.walk(directory_path):
+        for filename in filenames:
+            # Ignore files with certain extensions.
+            if not any(
+                    filename.endswith(p) for p in FILE_EXTENSIONS_TO_IGNORE):
+                total_file_count += 1
+    return total_file_count
+
+
+def _compare_file_count(source_path, target_path):
+    """Ensure that two dir's file counts match.
+
+    Args:
+       source_path: str. Source of staging files.
+       target_path: str. Final filepath or built file's directory.
+
+    Raises:
+        ValueError: The source directory does not have the same file count as
+            the target directory.
     """
-    f = open(source_path, 'r')
-    content = f.read()
+    source_dir_file_count = get_file_count(source_path)
+    target_dir_file_count = get_file_count(target_path)
+    if source_dir_file_count != target_dir_file_count:
+        print 'Comparing %s vs %s' % (source_path, target_path)
+        raise ValueError(
+            '%s files in source dir != %s files in target dir.' % (
+                source_dir_file_count, target_dir_file_count))
+
+
+def _match_filename_with_hashes(relative_filepath, file_hashes):
```

Comment:
Let's call this _verify_filepath_hash to make clear that this is a validation function, and put it together with the verify function lower down.

Reply:
Function renamed and relocated.